python/llm/src/bigdl/llm/transformers/models/llama.py

```diff
@@ -506,8 +505,18 @@ def llama_attention_selective_batching_forward_4_31(
     return attn_output.to(original_dtype), attn_weights, updated_past_key_values
 
 
-def check_flash_attention_available(query):
+def use_flash_attention(query):
+    bsz, q_len, _ = query.size()
     # check whether ipex flash attention can be used
+    if bsz > 1:
+        # only use flash attention for batch_size = 1 now
+        # as flash attention doesn't support attn_mask in ipex 2.1,
+        # so it will cause output error for padded batch input
+        return False
+    if q_len == 1:
```

Comment:
It seems flash attention can benefits the fp16 case even for the rest token when I experiment it with AutoTP.

Reply:
Thanks for reminder! I will take some further more test about flash attention.